packages/bsky/src/api/app/bsky/actor/searchActorsTypeahead.ts

```diff
@@ -82,10 +82,14 @@ const hydration = async (
 }
 
 const noBlocks = (inputs: RulesFnInput<Context, Params, Skeleton>) => {
-  const { ctx, skeleton, hydration } = inputs
-  skeleton.dids = skeleton.dids.filter(
-    (did) => !ctx.views.viewerBlockExists(did, hydration),
-  )
+  const { ctx, skeleton, hydration, params } = inputs
+  skeleton.dids = skeleton.dids.filter((did) => {
+    // Always display exact matches so that users can find profiles that they have blocked
+    const actor = hydration.actors?.get(did)
+    if (!actor) return false
+    const isExactMatch = actor.handle?.toLowerCase() === params.q?.toLowerCase()
```

Comment:
Two minor things to poke at here:
 - we're searching on `params.q ?? params.term` rather than `params.q`.
 - relatedly, if the actor has an invalid handle (missing `actor.handle`) and `params.q` is not set, blocked users with invalid handles will make it through.  That could happen in empty searches (though I think these are incidentally disallowed within the search service), or any search that uses `params.term` rather than `params.q`.

Reply:
Just curious, is #1 a backwards compat thing?

Reply:
I assume so— I think at some point @bnewbold standardized all the search query endpoints on `q` rather than `term`.  Either way, I figure we should either drop support for `term` or fully support it, and it looks like we're still supporting it today.

Reply:
yep sg! missed that, will push something up in a moment.